cmd/ddev/cmd/dotenv-set.go

```diff
@@ -72,6 +75,8 @@ ddev dotenv set .ddev/.env.redis --redis-tag 7-bookworm`,
 				envMap[envName] = value
 				changedEnvMap[envName] = value
 				hasUnknownFlags = true
+			} else {
+				util.Failed("The flag must be in long format, but received %s", flag)
```

Comment:
And I fail too here, so users can see that they passed a wrong flag to the command.

I don't think changing the rules would break this behavior because I'm already using the strictest check I could think of.